### PR TITLE
Don't tell Helm to 'keep' node pool resources

### PR DESCRIPTION
**Pull request by fiunchinho**
### What this PR does / why we need it
Towards https://github.com/giantswarm/giantswarm/issues/29061

### Checklist

- [X] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- There is no need to annotate node pool manifests with `"helm.sh/resource-policy": keep`, because that's the first thing that CAPI will remove anyway.
```
Comment (AndiDog) [inline CHANGELOG.md:12]: I don't quite understand this change log. We should state we fix deletion of node pools (and thereby also renames, but no need to mention renames).

**Comment by fiunchinho** [comment]
Added `skip/ci` label as e2e tests already passed twice for these changes.

**Review by AndiDog — APPROVED** [review]

**Review by calvix — APPROVED** [review]

**Pull request merged**